### ISSUE TITLE
TypeError: Cannot read property '1' of undefined

### ISSUE THREAD

**Issue by zfi** [parallaxinc/solo#490]
Reported by Sentry
TypeError init(src/modules/blockly/generators/propc/gpio)
error:
Cannot read property '1' of undefined

```
./src/modules/blockly/generators/propc/gpio.js in init at line 1519:22

    const profile = getDefaultProfile();
    this.setTooltip(Blockly.MSG_AB_VOLT_IN_TOOLTIP);
    this.setColour(colorPalette.getColor('io'));
    this.appendDummyInput()
        .appendField('A/D channel')
        .appendField(new Blockly.FieldDropdown(  <<=====
            profile.analog), 'CHANNEL')
        .appendField('read (0-5V) in volt-100ths');
    this.setOutput(true, 'Number');
    this.setPreviousStatement(false, null);
    this.setNextStatement(false, null);
```

**Comment by zfi** [comment]
Sentry is also reporting this at ./src/modules/blockly/generators/propc/sensors.js in init at line 170:22. These both share a common characteristic in that profile.analog is passed into the FieldDropdown() method.  profile.analog is a static array of [string, string]. Also, the only profile that contains a non-null analog array is the activity board profile. In order to generate the error reported in this issue, one of the following must be true:
* The profile or the profile.array is undefined.
* The board type is not an activity board
* The profile.array is getting truncated in code somewhere.

```
}
    this.setTooltip(Blockly.MSG_JOYSTICK_INPUT_TOOLTIP);
    this.setColour(colorPalette.getColor('input'));
    this.appendDummyInput()
        .appendField('Joystick ' + this.chan[0] + '-axis A/D')
        .appendField(new Blockly.FieldDropdown(  <<======
            profile.analog), 'PIN' + this.chan[1]);
    this.setOutput(true, 'Number');
    this.setPreviousStatement(false, null);
    this.setNextStatement(false, null);
  },
```

**Comment by zfi** [comment]
Issue corrected in v1.5.2
